tests/splines/non_periodic_spline_builder.cpp

```diff
@@ -179,6 +179,61 @@ TEST(NonPeriodicSplineBuilderTest, Identity)
     spline_evaluator
             .deriv(spline_eval_deriv.span_view(), coords_eval.span_cview(), coef.span_cview());
 
+    ddc::Chunk integral(spline_builder.batch_domain(), ddc::HostAllocator<double>());
+    spline_evaluator.integrate(integral.span_view(), coef.span_cview());
+
+    ddc::Chunk<double, ddc::DiscreteDomain<ddc::Deriv<typename IDimX::continuous_dimension_type>>>
+            quadrature_coefficients_derivs_xmin;
+    ddc::Chunk<double, ddc::DiscreteDomain<IDimX>> quadrature_coefficients;
+    ddc::Chunk<double, ddc::DiscreteDomain<ddc::Deriv<typename IDimX::continuous_dimension_type>>>
+            quadrature_coefficients_derivs_xmax;
+    std::
+            tie(quadrature_coefficients_derivs_xmin,
+                quadrature_coefficients,
+                quadrature_coefficients_derivs_xmax)
+            = spline_builder.quadrature_coefficients();
+#if defined(BCL_HERMITE)
+    double const quadrature_integral_derivs_xmin = ddc::parallel_transform_reduce(
+            Kokkos::DefaultHostExecutionSpace(),
+            quadrature_coefficients_derivs_xmin.domain(),
+            0.0,
+            ddc::reducer::sum<double>(),
+            [&](ddc::DiscreteElement<ddc::Deriv<typename IDimX::continuous_dimension_type>> const
+                        ix) {
+                ddc::Coordinate<DimX> const dx
+                        = ddc::distance_at_right(interpolation_domain.front() + 1);
+                return quadrature_coefficients_derivs_xmin(ix)
+                       * (*deriv_l)(ix)*ddc::detail::ipow(dx, ix.uid());
+            });
+#else
+    double const quadrature_integral_derivs_xmin = 0.;
+#endif
+    double quadrature_integral = ddc::parallel_transform_reduce(
+            Kokkos::DefaultHostExecutionSpace(),
+            quadrature_coefficients.domain(),
+            0.0,
+            ddc::reducer::sum<double>(),
+            [&](ddc::DiscreteElement<IDimX> const ix) {
+                return quadrature_coefficients(ix) * yvals(ix);
+            });
+#if defined(BCR_HERMITE)
+    double const quadrature_integral_derivs_xmax = ddc::parallel_transform_reduce(
+            Kokkos::DefaultHostExecutionSpace(),
+            quadrature_coefficients_derivs_xmax.domain(),
+            0.0,
+            ddc::reducer::sum<double>(),
+            [&](ddc::DiscreteElement<ddc::Deriv<typename IDimX::continuous_dimension_type>> const
+                        ix) {
+                ddc::Coordinate<DimX> const dx
+                        = ddc::distance_at_left(interpolation_domain.back() - 1);
+                return quadrature_coefficients_derivs_xmax(ix)
+                       * (*deriv_r)(ix)*ddc::detail::ipow(dx, ix.uid());
```

Comment:
@EmilyBourne the term `ddc::detail::ipow(dx, ix.uid())` will always appear on the user side. Shall it be directly included in `quadrature_coefficients_derivs_xmax` ? Are there situations where you only need the current `quadrature_coefficients_derivs_xmax` ?

Reply:
Yeah good question, I would prefer to put `* (*deriv_r)(ix)*ddc::detail::ipow(dx, ix.uid())` inside `quadrature_coefficients_derivs_xmax(ix)`

Reply:
I agree that this should appear in `quadrature_coefficients_derivs_xmax`. This factor is included to improve the condition number of the matrix. Once the matrix equation is solved there is no reason to keep this trick

Reply:
[38c2c0c](https://github.com/CExA-project/ddc/pull/560/commits/38c2c0cae61f1a77ecaf3fc453dbea451edf06d6)